File: app/src/main/java/me/eagzzycsl/darkroom/ui/AdapterOnDeviceApp.java
package me.eagzzycsl.darkroom.ui;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import me.eagzzycsl.darkroom.R;
import me.eagzzycsl.darkroom.model.OnDeviceApp;

class AdapterOnDeviceApp extends BaseAdapter<OnDeviceApp> {
    AdapterOnDeviceApp(Context context) {
        super(context);
    }

    @Override
    public int getLayoutId() {
        return R.layout.layout_item_on_device;
    }

    @Override
    public UserHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new UserHolder(LayoutInflater.from(getContext()).inflate(getLayoutId(), parent, false));
    }

    private class UserHolder extends RecViewHolder<OnDeviceApp> implements View.OnClickListener {
        private final TextView item_name;
        private final ImageView item_icon;


        UserHolder(View itemView) {
            super(itemView);
            item_name = itemView.findViewById(R.id.item_name);
            item_icon = itemView.findViewById(R.id.item_icon);
            itemView.setOnClickListener(this);

        }

        @Override
        public void setContent(OnDeviceApp onDeviceApp) {
            item_name.setText(onDeviceApp.getAppName());
            item_icon.setImageDrawable(onDeviceApp.getAppIcon());
            if (onDeviceApp.getFrozen()) {
                item_name.setTextColor(getContext().getColor(R.color.colorPrimary));
            } else {
                item_name.setTextColor(getContext().getColor(android.R.color.black));
            }
            itemView.setTag(onDeviceApp);
        }

        @Override
        public void onClick(View view) {
            OnDeviceApp onDeviceApp = (OnDeviceApp) view.getTag();
            onDeviceApp.toggleFrozen();
            AdapterOnDeviceApp.this.notifyItemChanged(this.getLayoutPosition());
        }
    }

}

File: app/src/main/java/me/eagzzycsl/darkroom/ui/AdapterOnDeviceApp.kt
package me.eagzzycsl.darkroom.ui

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import me.eagzzycsl.darkroom.R
import me.eagzzycsl.darkroom.model.MetaApp

internal class AdapterOnDeviceApp(context: Context) : BaseAdapter<MetaApp>(context) {

    override val layoutId: Int
        get() = R.layout.layout_item_on_device

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OnDeviceHolder {
        return OnDeviceHolder(LayoutInflater.from(context).inflate(layoutId, parent, false))
    }

    inner class OnDeviceHolder internal constructor(itemView: View) : BaseAdapter<MetaApp>.RecViewHolder<MetaApp>(itemView), View.OnClickListener {
        private val itemName: TextView = itemView.findViewById(R.id.item_name)
        private val itemIcon: ImageView = itemView.findViewById(R.id.item_icon)


        init {
            itemView.setOnClickListener(this)
        }

        override fun setContent(data: MetaApp) {
            itemName.text = data.appName
            itemIcon.setImageDrawable(data.appIcon)
            if (data.willFreeze) {
                itemName.setTextColor(context.getColor(R.color.colorPrimary))
            } else {
                itemName.setTextColor(context.getColor(android.R.color.black))
            }
            itemView.tag = data
        }

        override fun onClick(view: View) {
            val onDeviceApp = view.tag as MetaApp
            onDeviceApp.toggleWillFreeze()
            this@AdapterOnDeviceApp.notifyItemChanged(this.layoutPosition)
        }
    }

}
